### PR TITLE
Create Web Application Deception Technology

### PR DESCRIPTION
**Pull request by vabmalikusa**
Deception technology is an emerging approach to cybersecurity that involves distributing decoy systems, credentials, and data throughout an environment to deceive, detect, and disrupt attackers. Rather than simply blocking all malicious activity, deception techniques aim to mislead attackers and waste their time while providing high-fidelity alerts of their presence.

### CODE REVIEW

File: pages/controls/Web Application Deception Technology [hunk 1]
```diff
@@ -0,0 +1,78 @@
+---
+
+layout: col-sidebar
+title: Web Application Deception Technology
+author: Vaibhav Malik
+contributors:
+permalink: [direct link to page, removes /pages] (this is optional and requires some care)
+tags: [attack, XSS, etc]
```
Comment (kingthorin) [inline pages/controls/Web Application Deception Technology:8]: Please adjust this, look at others for guidance if you need

Comment (vabmalikusa) [reply pages/controls/Web Application Deception Technology]: Done

File: pages/controls/Web Application Deception Technology [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+---
+
+ {% include writers.html %}
+
+
+Description
```
[the remark this answers is not in the document]
Comment (vabmalikusa) [reply pages/controls/Web Application Deception Technology]: Done

File: pages/controls/Web Application Deception Technology [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+Deception technology is an emerging approach to cybersecurity that involves distributing decoy systems, credentials, and data throughout an environment to deceive, detect, and disrupt attackers. Rather than simply blocking all malicious activity, deception techniques aim to mislead attackers and waste their time while providing high-fidelity alerts of their presence.
```
[the remark this answers is not in the document]
Comment (vabmalikusa) [reply pages/controls/Web Application Deception Technology]: A little different but addedhttps://www.rapid7.com/fundamentals/deception-technology/

File: pages/controls/Web Application Deception Technology [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+For web application security, deception can take several forms:
+
+1. Honeytoken Cookies - Fake session cookies containing enticing names (e.g., "admin_session"), but invalid values are seeded into the application. Any attempts to use these cookies indicate an attacker is stealing and replaying cookies.
```
[the remark this answers is not in the document]
Comment (vabmalikusa) [reply pages/controls/Web Application Deception Technology:20]: Makes sense, added few more

File: pages/controls/Web Application Deception Technology [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+2. Deceptive Links/URLs - Hidden links and API endpoints are placed in the application's client-side code. Regular users never see these links, so any requests to them suggest an attacker is inspecting the source code for vulnerabilities.  
+
+3. Booby-trapped Form Fields—Invisible form fields are added to lure attackers. These fields are not rendered for real users, so any data submitted in them points to malicious form tampering.
```
[the remark this answers is not in the document]
Comment (vabmalikusa) [reply pages/controls/Web Application Deception Technology]: Done

File: pages/controls/Web Application Deception Technology [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+
+4. Canary Files/Records - Fake sensitive files (e.g., "salaries.xlsx") or database records are planted in the application. Access to these dummy assets indicates an attacker has broken in and is snooping around.
+
+5. Tarpit Pages - Requests for certain pages (e.g., WordPress's wp-admin for a non-WordPress site) are redirected to an infinite loop or a resource-intensive puzzle. This slows down attackers, especially scanners and crawlers.
+
+Deception technology provides several advantages for web application defenders:
+- High signal-to-noise ratio—Alerts from deception systems are highly confident since legitimate users rarely trigger them. This reduces alert fatigue for security teams.
```
[the remark this answers is not in the document]
Comment (vabmalikusa) [reply pages/controls/Web Application Deception Technology]: High signal-to-noise ratio is seperate. Fixed

File: pages/controls/Web Application Deception Technology [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+- Early detection - Deceptions can catch attackers early in the reconnaissance and lateral movement stages before they cause damage.
+- Attribution - Analyzing how attackers interact with deceptions can reveal their intent, tools, and techniques, helping to profile the adversary.
+Threat intelligence - Collecting data on attacker behavior can generate indicators of compromise (IOCs) and threat intelligence to strengthen other defenses.
```
[the remark this answers is not in the document]
Comment (vabmalikusa) [reply pages/controls/Web Application Deception Technology:38]: Done

File: pages/controls/Web Application Deception Technology [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+3. Booby-trapped Form Fields—Invisible form fields are added to lure attackers. These fields are not rendered for real users, so any data submitted in them points to malicious form tampering.
+
+4. Canary Files/Records - Fake sensitive files (e.g., "salaries.xlsx") or database records are planted in the application. Access to these dummy assets indicates an attacker has broken in and is snooping around.
+
+5. Tarpit Pages - Requests for certain pages (e.g., WordPress's wp-admin for a non-WordPress site) are redirected to an infinite loop or a resource-intensive puzzle. This slows down attackers, especially scanners and crawlers.
+
+Deception technology provides several advantages for web application defenders:
+- High signal-to-noise ratio—Alerts from deception systems are highly confident since legitimate users rarely trigger them. This reduces alert fatigue for security teams.
+- Early detection - Deceptions can catch attackers early in the reconnaissance and lateral movement stages before they cause damage.
+- Attribution - Analyzing how attackers interact with deceptions can reveal their intent, tools, and techniques, helping to profile the adversary.
```
[the remark this answers is not in the document]
Comment (vabmalikusa) [reply pages/controls/Web Application Deception Technology]: Done

File: pages/controls/Web Application Deception Technology [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+
+3. Honeytoken API Keys Fake API keys or access tokens can be seeded in an application's JavaScript or HTML source. Attackers who scrape the client-side code looking for sensitive tokens to steal will pick up these fake tokens. Any use of these honeytoken API keys indicates malicious recon activity.
+
+4. Deceptive Links/URLs - Hidden links and API endpoints are placed in the application's client-side code. Regular users never see these links, so any requests to them suggest an attacker is inspecting the source code for vulnerabilities.  
+
+5. Booby-trapped Form Fields—Invisible form fields are added to lure attackers. These fields are not rendered for real users, so any data submitted in them points to malicious form tampering.
+
+6. Canary Files/Records - Fake sensitive files (e.g., "salaries.xlsx") or database records are planted in the application. Access to these dummy assets indicates an attacker has broken in and is snooping around.
+
+7. Tarpit Pages - Requests for certain pages (e.g., WordPress's wp-admin for a non-WordPress site) are redirected to an infinite loop or a resource-intensive puzzle. This slows down attackers, especially scanners and crawlers.
```
Comment (kingthorin) [inline pages/controls/Web Application Deception Technology:32]:
```suggestion
1. Honeytoken Cookies - Fake session cookies containing enticing names (e.g., "admin_session"), but invalid values are seeded into the application. Any attempts to use these cookies indicate an attacker is stealing and replaying cookies.
2. Honeytoken Form Fields — In addition to hidden form fields, honeytokens can be placed in visible fields that normal users should never use. For example, an "admin_override" field that regular users have no reason to interact with. Any submissions with that field populated indicate an attacker probing the application.
3. Honeytoken API Keys - Fake API keys or access tokens can be seeded in an application's JavaScript or HTML source. Attackers who scrape the client-side code looking for sensitive tokens to steal will pick up these fake tokens. Any use of these honeytoken API keys indicates malicious recon activity.
4. Deceptive Links/URLs - Hidden links and API endpoints are placed in the application's client-side code. Regular users never see these links, so any requests to them suggest an attacker is inspecting the source code for vulnerabilities.  
5. Booby-trapped Form Fields - Invisible form fields are added to lure attackers. These fields are not rendered for real users, so any data submitted in them points to malicious form tampering.
6. Canary Files/Records - Fake sensitive files (e.g., "salaries.xlsx") or database records are planted in the application. Access to these dummy assets indicates an attacker has broken in and is snooping around.
7. Tarpit Pages - Requests for certain pages (e.g., WordPress's wp-admin for a non-WordPress site) are redirected to an infinite loop or a resource-intensive puzzle. This slows down attackers, especially scanners and crawlers.
```

File: pages/controls/Web Application Deception Technology [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+Attribution - Analyzing how attackers interact with deceptions can reveal their intent, tools, and techniques.
+Threat intelligence - Collecting data on attacker behavior can generate indicators of compromise (IOCs) and threat intelligence to strengthen other defenses.
+
+However, deception technology also has some challenges and considerations:
+Maintenance - Deceptions must be kept fresh and believable. Attackers may grow suspicious if they repeatedly encounter stale or unconvincing decoys.  
+Interference - Deceptions should not interfere with normal application functionality or confuse legitimate users.
+Legal issues - Consult with legal counsel on entrapment and privacy laws before implementing deceptions, especially in public-facing systems.
+
+## Sample Code
+Below is a Python snippet using the Flask web framework to create a web application honeypot with a deceptive link:
```
Comment (kingthorin) [inline pages/controls/Web Application Deception Technology:46]:
```suggestion
## Sample Code

Below is a Python snippet using the Flask web framework to create a web application honeypot with a deceptive link:
```

File: pages/controls/Web Application Deception Technology [hunk 11]
```diff
@@ -0,0 +1,82 @@
+---
+
+layout: col-sidebar
+title: Web Application Deception Technology
+author: Vaibhav Malik
+tags: [controls, application, web]
+permalink: /controls/Web Application Deception
```
Comment (kingthorin) [inline pages/controls/Web Application Deception Technology:7]:
```suggestion
permalink: /controls/WebAppDeception
```

**Review by kingthorin — CHANGES_REQUESTED** [review]

**Review by kingthorin — APPROVED** [review]

**Comment by kingthorin** [comment]
Thanks!

**Pull request merged**